modules/signals/spec/signal-method.spec.ts

```diff
@@ -72,18 +72,15 @@ describe('signalMethod', () => {
       const adder = createAdder((value) => (a += value));
       adder(summand1);
       adder(summand2);
-
-      summand1.set(2);
-      summand2.set(3);
       TestBed.flushEffects();
-      expect(a).toBe(6);
+      expect(a).toBe(4);
```

Comment:
I was doing some testing and it didn't work the way as I expected when an effect is destroyed manually.
Is this the desired behavior?

```ts
it('stops specific tracking when calling destroy manually on an effect', () => {
      let a = 1;
      const summand1 = signal(1);
      const summand2 = signal(2);
      const adder = createAdder((value) => (a += value));
      adder(summand1);

      const s2 = adder(summand2);
      // 👇 immediately destroy
      s2.destroy();

      TestBed.flushEffects();
      expect(a).toBe(4);

      summand1.set(100);
      summand2.set(3000);

      TestBed.flushEffects();

      // 👇 Fails, received 3104
      expect(a).toBe(1004); 
    });
```

Reply:
Ouch, that hurts. Thanks Tim, let me fix that as well.

Reply:
Ok, now it starts to become interesting. That is a behavior of `effect`. That test also fails:

```typescript
  it('should destroy an effect', () => {
    let effectCounter = 0;
    const counter = signal(1);
    const effectRef = TestBed.runInInjectionContext(() =>
      effect(() => {
        counter();
        effectCounter++;
      }, { injector: TestBed.inject(EnvironmentInjector) })
    );
    expect(effectCounter).toBe(0);
    effectRef.destroy();
    TestBed.flushEffects();
    expect(effectCounter).toBe(0);
  });
  ```

Reply:
@timdeschryver, I've now added a test, that destroys the instance manually after the first `flushEffects()` and that one works as expected.

I have been passed on the behavior of a non working destroy before the first flush but it is something that needs to be fixed by Angular.

Therefore, I'd suggest that this one is seen as resolved.

Reply:
Thanks for testing this @rainerhahnekamp.
I agree that this is then the expected Angular behavior.